sys-process/btop: Backport clang fix

vimproved opened the pull request:
Bug: https://bugs.gentoo.org/839318
Closes: https://bugs.gentoo.org/908670

The same commit that allowed building with clang also fixes https://bugs.gentoo.org/908670 (musl LFS64 issue).

NexAdn commented on sys-process/btop/files/btop-1.2.13-allow-clang.patch:
```diff
@@ -0,0 +1,234 @@
+Bug: https://bugs.gentoo.org/839318
+Upstream Commit: https://github.com/aristocratos/btop/commit/7e50b03e391a346b2b6ce92be332c58d91f75a000w
+
+From af7d5086fce8d995f3f8b10677d107a1124a12bf Mon Sep 17 00:00:00 2001
+From: "Jakob P. Liljenberg" <admin@qvantnet.com>
+Date: Thu, 15 Jun 2023 17:45:05 +0200
+Subject: [PATCH] Merge pull request #510 from nobounce/LLVM
```
This info doesn't really need to be in the patch file in the tree, does it?

vimproved replied:
Probably not. scrub-patch didn't remove it so I didn't realize it was still there.

thesamesam replied:
Which part? This looks fine.

thesamesam replied:
I don't think this change was right. Having the metadata is important. You can keep the headers and just say "(with hunks from)" or something if you're modifying it.

NexAdn commented on sys-process/btop/files/btop-1.2.13-allow-clang.patch:
```diff
@@ -0,0 +1,234 @@
+ 		}
+-		current_boxes = move(new_boxes);
++		current_boxes = std::move(new_boxes);
+ 		return true;
+ 	}
+ 
+--- a/src/btop_tools.cpp
++++ b/src/btop_tools.cpp
+@@ -17,6 +17,7 @@ tab-size = 4
+ */
+ 
+ #include <cmath>
++#include <codecvt>
+ #include <iostream>
+ #include <fstream>
+ #include <ctime>
+@@ -151,7 +152,7 @@ namespace Term {
+ 
+ //? --------------------------------------------------- FUNCTIONS -----------------------------------------------------
+ 
+-// ! Dsiabled due to issue when compiling with musl, reverted back to using regex
++// ! Disabled due to issue when compiling with musl, reverted back to using regex
+ // namespace Fx {
+ // 	string uncolor(const string& s) {
+ // 		string out = s;
+--- a/src/linux/btop_collect.cpp
++++ b/src/linux/btop_collect.cpp
+@@ -1079,9 +1079,9 @@ namespace Mem {
+ 				bool new_ignored = false;
+ 				for (auto& [mountpoint, disk] : disks) {
+ 					if (std::error_code ec; not fs::exists(mountpoint, ec) or v_contains(ignore_list, mountpoint)) continue;
+-					struct statvfs64 vfs;
+-					if (statvfs64(mountpoint.c_str(), &vfs) < 0) {
+-						Logger::warning("Failed to get disk/partition stats for mount \""+ mountpoint + "\" with statvfs64 error code: " + to_string(errno) + ". Ignoring...");
++					struct statvfs vfs;
++					if (statvfs(mountpoint.c_str(), &vfs) < 0) {
++						Logger::warning("Failed to get disk/partition stats for mount \""+ mountpoint + "\" with statvfs error code: " + to_string(errno) + ". Ignoring...");
+ 						ignore_list.push_back(mountpoint);
+ 						new_ignored = true;
+ 						continue;
```
This part of the patch doesn't really apply to the description “allow clang”. Also, I don't see a reason for all of these changes to be backported. For the sake of keeping the patch short and simple, please reduce the changes to what is necessary to make the clang build and the musl build work.

vimproved replied:
I'll go ahead and split the patch into one for musl and one for allowing compilation with clang 16. I'm pretty sure all of the changes are needed for either clang or musl, but I'll go ahead and make sure. It was initially in one patch because it was all one commit.

thesamesam replied:
Yes, this will be needed for musl-1.2.4 (statvfs64).

NexAdn commented on sys-process/btop/btop-1.2.13-r2.ebuild:
```diff
@@ -0,0 +1,65 @@
+	"${FILESDIR}/${P}-fix-makefile-deps.patch"
+	"${FILESDIR}/${P}-verbose-mkdir.patch"
+
+	# Backported fix for compilation with clang 16.
+	# can be removed in 1.2.14 or later
+	"${FILESDIR}/${P}-allow-clang.patch"
```
Since this patch does a bit more than just add support for Clang 16,
this description is a bit unsuited.
Maybe expand the description to include the note about the backported musl fix and links to the related bugs?

NexAdn reviewed (changes requested):
I'm okay with the overall changes. I have a few remarks regarding cosmetic changes, readability/understandability and simplifying the patches. It'd be nice if you could have a look at those.

vimproved commented on sys-process/btop/btop-1.2.13-r2.ebuild:
```diff
@@ -0,0 +1,65 @@
+
+pkg_setup() {
+	if [[ "${MERGE_TYPE}" != "binary" ]]; then
+			if tc-is-clang ; then
```
Not sure how two levels of indentation got there in the first place, sorry!

vimproved commented:
Split into two patches / commits, and resolved all the issues you brought up in review. Removed a few non-crucial parts of the patch for the sake of size, such as silencing some warnings from clang.

vimproved commented:
Re-added metadata.

NexAdn reviewed (approved):
LGTM. Thanks for adjusting the PR!

nobounce commented:
Why would you think [this](https://github.com/aristocratos/btop/commit/a5e4969d7e6109da8241f04db8ef2fc626c4559d) is a non crucial part of the diff?

vimproved commented:
> Why would you think [this](https://github.com/aristocratos/btop/commit/a5e4969d7e6109da8241f04db8ef2fc626c4559d) is a non crucial part of the diff?

ah sorry, didn't see the reasoning for that. I'll re-add it.
